pkg/api/customization/cluster/actions_etcd.go

```diff
@@ -106,8 +112,32 @@ func (a ActionHandler) RestoreFromEtcdBackupHandler(actionName string, action *t
 		}
 	}
 
+	if input.RestoreRkeConfig != "" && backup.Status.ClusterObject == "" {
```

Comment:
Aren't you potentially incorrectly disallowing the kubeVersion part below since it doesn't use the status.clusterObject ?

Reply:
both version and whole config are optional, pass in "" or no param at all if you want the original feature. that param not passed is original behavior, pass "all' you get whole rke config, pass "kubernetesVersion" for restoring just the vk8s version

Reply:
also this is checking to see if you are trying to restore the version/config AND the backup was taken before this snapshot feature was upgraded